### PR TITLE
Add gate validation

### PR DESCRIPTION
**Pull request by AthenaCaesura**
## Description

Ensure that `QuantumPrograms` that have both T gates and rotation gates cannot pass though the pipeline unless the rotation gates are decomposed. Ran into this issue when performing resource estimates for hamiltonians, so I figured I would fix it now.

## Please verify that you have completed the following steps

- [x] I have self-reviewed my code.
- [x] I have included test cases validating introduced feature/fix.
- [ ] I have updated documentation.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -19,27 +19,27 @@ include_package_data = True
...
+python_requires = >=3.9,!=3.9.7,<3.11
 
 install_requires =
-    orquestra-quantum==0.9.0
-    orquestra-qiskit==0.9.0
-    orquestra-cirq==0.8.0
+    orquestra-quantum~=0.9.0
+    orquestra-qiskit~=0.9.0
+    # temporary hack until newest version of orquestra-cirq is released
+    orquestra-cirq @ git+https://github.com/zapatacomputing/orquestra-cirq.git
```
Comment (mstechly) [inline setup.cfg:28]: Please pin to a specific commit, this might change whenever someone pushes to `main`.

File: setup.cfg [hunk 2]
```diff
@@ -19,27 +19,27 @@ include_package_data = True
...
     # Used to define and run Orquestra workflows
     orquestra-sdk[all]==0.45.1
     matplotlib>=3.6
     numpy>=1.20
     more-itertools~=9.1.0
     pandas==1.5.3
-    # pyLIQTR was forked to remove dependencies on cirq-rigetti which has conflicting
-    # dependencies with ray (used by orquestra-sdk)
-    pyLIQTR @ git+https://github.com/SebastianMorawiec/pyLIQTR_test.git@fa3a0e410ff1cf34d6d060118268a319ac793604
+    pyLIQTR~=0.3.0
```
Comment (mstechly) [inline setup.cfg:36]: Same here.

File: tests/benchq/resource_estimation/juliapkg.json
Comment (mstechly) [inline tests/benchq/resource_estimation/juliapkg.json:1]: not the right place to put this file.

**Review by mstechly — APPROVED** [review]

File: setup.cfg (hunk 2 above)
Comment (AthenaCaesura) [reply setup.cfg:36]: Wait, isn't it being tagged to a version good enough?

File: tests/benchq/resource_estimation/juliapkg.json
Comment (AthenaCaesura) [reply tests/benchq/resource_estimation/juliapkg.json:1]: will remove. 🤦‍♀️

File: setup.cfg (hunk 1 above)
Comment (AthenaCaesura) [reply setup.cfg:28]: I think I'm just going to wait until we publish the latest version of `orquestra-cirq` to finish this PR.

**Pull request closed without merging**